### PR TITLE
Ticket4115 add ioc oercone

### PR DESCRIPTION
**Pull request by JamesKingWork**
### Description of work

Add an OPI to track the pressure of an Oerlikon Centre-One pressure gauge

### Ticket

[ISISComputingGroup/IBEX#4115](https://github.com/ISISComputingGroup/IBEX/issues/4115)

### Acceptance criteria

1. I can view the pressure changing over time in a graph
2. I can view the current pressure in a value
3. I can view and set the units of the pressure

### Unit tests

None, OPI only

### System tests

None, OPI only

### Documentation

https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OERCONE

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.opis/resources/.opi [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  <show_edit_range>true</show_edit_range>
+  <show_grid>true</show_grid>
+  <show_ruler>true</show_ruler>
+  <snap_to_geometry>true</snap_to_geometry>
+  <widget_type>Display</widget_type>
+  <width>800</width>
+  <wuid>5611881a:16cb46ec964:-7f3c</wuid>
+  <x>-1</x>
+  <y>-1</y>
+</display>
```
Comment (Tom-Willemsen) [inline base/uk.ac.stfc.isis.ibex.opis/resources/.opi]: This file looks like a mistake as it is just called `.opi`.

If it's needed can you explain why?

File: base/uk.ac.stfc.isis.ibex.opis/resources/opi_info.xml [hunk 2]
```diff
@@ -2159,11 +2159,29 @@
...
         <description>The OPI for the NIMROD Gas Panel</description>
-        <macros />
+        <macros/>
         <categories>
           <category>Gas and liquid handling systems</category>
         </categories>
       </value>
     </entry>
+    <entry>
+      <key>NIMROD Oerlikon Centre-One Pressure Gauge</key>
```
Comment (Tom-Willemsen) [inline base/uk.ac.stfc.isis.ibex.opis/resources/opi_info.xml]: I'm not sure this is unique to NIMROD - usually unless it's truly fixed (e.g. jaw sets or fully custom gas panels etc) we would just call this "Oerlikon Centre-One"

File: base/uk.ac.stfc.isis.ibex.opis/resources/.opi (hunk 1 above)
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.opis/resources/.opi]: Yes I don't remember adding this, removing it now

File: base/uk.ac.stfc.isis.ibex.opis/resources/opi_info.xml (hunk 2 above)
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.opis/resources/opi_info.xml]: Amended to not mention nimrod in record

**Comment by JamesKingWork** [comment]
> 
> 
> The OPI isn't displaying nicely in the GUI for me - there seems to be excessive amounts of spacing
> 
> See screenshot
> image
> 
> I also can't get the OPI to connect as PV_ROOT isn't defined.
> 
> Did you forget to push a commit?

Spacing has now been removed and PV_ROOT is now set, I'm not sure how it managed to connect before unless the value got removed somehow, but should be fine now

**Comment by Tom-Willemsen** [comment]
The OPI still isn't looking right for me:

image

- The graph isn't displaying
- The units are displaying as an integer rather than a string (this is an option you can set in CS-Studio)

**Pull request merged**